### ISSUE TITLE
Calling .mock with a function as a response is not receiving request object

### ISSUE THREAD

**Issue by aaronschwartz** [wheresrhys/fetch-mock#390]
Environment: 
fetch-mock@7.2.2
webpack@4
whatwg-fetch (installed as 'entry' in 'webpack.config')

I'm trying to receive the Request object so that I can send a response based on the request body.

```
// From the documentation, there should be a request object included as the third parameter
.mock('/example', (url, opts, request) => return request.json().then(...))
```

`url` and `opts` are being filled in correctly, but `request` is always `undefined`. I've read the documentation about configuring a custom Request object, but I've been unable to figure out what I'm supposed to configure it to. I've tried various combinations. 

e.g. (not working)
```
import {Request as myRequest} from 'whatwg-fetch';

fetchMock.config.Request = myRequest
```

I think the issue might actually be a bug in `fetch-mock`.

If I'm reading the code correctly, the function passed in is being called without the third argument. I'm also not seeing any test cases for this use case.

https://github.com/wheresrhys/fetch-mock/blob/master/src/lib/fetch-handler.js#L45

This scenario was doable in versions <= 6.xx before the request was parsed and split into different parts.

Thanks for the library, it's been really great.

**Comment by wheresrhys** [comment]
yep this is a bug, I found it the other day too, but not had time to fix it yet. Thanks for reporting

**Comment by wheresrhys** [comment]
This should be fixed in 7.2.4
